### PR TITLE
LPS-72641 SF

### PR DESCRIPTION
**Pull request by ealonso**

### CODE REVIEW

File: modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/js/TagSelector.es.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ */
+class TagSelector extends Component {
+	/**
+	 * @inheritDoc
+	 */
+	rendered() {
+		AUI().use(
+			'liferay-asset-taglib-tags-selector',
+			function(A) {
+				this.input_ = this.element.querySelector('input[type="text"]');
```
Comment (jbalsas) [inline modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/js/TagSelector.es.js]: This is going to be async and will depend on the load of the `liferay-asset-taglib-tags-selector` module. Better set this in the `attached` lifecycle

File: modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/js/TagSelector.es.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+					input: `#${this.input_.getAttribute('id')}`,
+					portletURL: this.tagSelectorURL,
+					tagNames: this.rule.queryValues || ''
+				};
+
+				this.tagsSelector_ = new Liferay.AssetTaglibTagsSelector(config);
+
+				const entries = this.tagsSelector_.entries;
+
+				this.element.addEventListener('click', this.focusTagInput_.bind(this));
```
Comment (jbalsas) [inline modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/js/TagSelector.es.js]: This should be done in the `attached` lifecycle as well.

File: modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/js/TagSelector.es.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+				entries.after('add', this.onEntriesChanged_, this);
+				entries.after('remove', this.onEntriesChanged_, this);
+
+				this.tagsSelector_.render();
+				this.element.parentNode.removeAttribute('tabindex');
```
Comment (jbalsas) [inline modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/js/TagSelector.es.js]: Why is this needed? Is the `TagsSelector` render adding it?

File: modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/js/CategorySelector.es.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+				this.categoriesSelector_ = new Liferay.AssetTaglibCategoriesSelector(config);
+
+				const entries = this.categoriesSelector_.entries;
+
+				entries.after('add', this.onEntriesChanged_, this);
+				entries.after('remove', this.onEntriesChanged_, this);
+
+				this.categoriesSelector_.render();
+				this.element.parentNode.removeAttribute('tabindex');
```
Comment (jbalsas) [inline modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/js/CategorySelector.es.js]: Why is this needed? Is the `CategoriesSelector` render adding it?

File: modules/apps/web-experience/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/js/asset_taglib_tags_selector.js [hunk 5]
```diff
@@ -273,7 +273,7 @@ AUI.add(
 										if (!serviceQueryObj) {
 											serviceQueryObj = {
 												end: 20,
-												groupIds: instance.get('groupIds'),
+												groupIds: instance.get('groupIds').filter(function (groupId) { return groupId; }),
```
Comment (jbalsas) [inline modules/apps/web-experience/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/js/asset_taglib_tags_selector.js]: Why is this needed? I'm guessing we have wrong data in there. If that's the case, the question should be how to ensure we don't. Otherwise, we will need hacks like this everywhere...

File: modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/js/TagSelector.es.js (hunk 3 above)
Comment (p2kmgcl) [inline modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/js/TagSelector.es.js]: While I was reviewing the behaviour of keyboard navigation, I saw that an a tabindex was added by the tag selector. This is a patch that should work until we stop using A.use(), otherwise I would need to check in what cases that tabindex should be added or not. Is that right? Note that keeping the tabindex is totally fine, but it adds an unnecessary stop while tabbing through elements.

File: modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/js/CategorySelector.es.js (hunk 4 above)
Comment (p2kmgcl) [inline modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/js/CategorySelector.es.js]: Same as TagsSelector (change before)

File: modules/apps/web-experience/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/js/asset_taglib_tags_selector.js (hunk 5 above)
Comment (p2kmgcl) [inline modules/apps/web-experience/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/js/asset_taglib_tags_selector.js]: I checked the array being sent to the server and it was an array with a single empty string ([""]). The server tries to cast it to integer and it crashes. I will ask for help in order to find where is this data being generated.

File: modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/js/TagSelector.es.js (hunk 1 above)
Comment (p2kmgcl) [inline modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/js/TagSelector.es.js]: 👍

File: modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/js/TagSelector.es.js (hunk 2 above)
Comment (p2kmgcl) [inline modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/js/TagSelector.es.js]: 👍

File: modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/js/TagSelector.es.js (hunk 3 above)
Comment (jbalsas) [inline modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/js/TagSelector.es.js]: It's fine for now, then

File: modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/js/CategorySelector.es.js (hunk 4 above)
Comment (jbalsas) [inline modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/js/CategorySelector.es.js]: 👍

**Comment by p2kmgcl** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by p2kmgcl** [comment]
#755
